mypy/fastparse.py

```diff
@@ -629,15 +639,18 @@ def visit_ClassDef(self, n: ast3.ClassDef) -> ClassDef:
                         metaclass=dict(keywords).get('metaclass'),
                         keywords=keywords)
         cdef.decorators = self.translate_expr_list(n.decorator_list)
-        if n.decorator_list and sys.version_info >= (3, 8):
-            # Before 3.8, n.lineno points to the first decorator; in
-            # 3.8, it points to the 'class' statement.  We always make
-            # it point to the first decorator.  (The node structure
-            # here is different than for decorated functions.)
-            cdef.line = n.decorator_list[0].lineno
-            cdef.column = n.col_offset
+        # It's okay if end_lineno is *before* lineno here.
+        # lineno is where the error is reported, and end_lineno defines the
+        # opposite end of the range to search for "# type: ignore" comments in.
+        # This reports the correct line in the event of an error, but keeps old
+        # pre-3.8 "# type: ignore" comments working correctly.
```

Comment:
Similar comment (except I wasn't really confused here, since I had just figured out the previous one :-).

Reply:
As above:

```py
# Set end_lineno to the old mypy 0.700 lineno, in order to keep
# existing "# type: ignore" comments working:
```